### DISCUSSION TITLE
Entities not detected with aliased namespace imports.

### DISCUSSION THREAD

**Discussion by aik099** [doctrine/doctrine2#8621] [Support Questions] [answered]
The entity class (generated mostly by DQL plugin of PhpStorm):

```php
<?php

namespace Company\Project\DoctrineEntity;

use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity(repositoryClass="Company\Project\DoctrineRepository\ContactUsFormSubmissionRepository")
 */
class ContactUsFormSubmission
{

    /**
     * @ORM\Id
     * @ORM\Column(type="integer")
     * @ORM\GeneratedValue
     */
    private $id;

    /**
     * @var string
     * @ORM\Column(type="string", length=30)
     */
    private $firstName;

    /**
     * @var string
     * @ORM\Column(type="string", length=30)
     */
    private $lastName;

}
```

Here is how Doctrine ORM is initiated:

```php
// Configure database.
$db_params = array(
   'driver' => 'pdo_mysql',
   'host' => $_SERVER['DB_HOST'],
   'dbname' => $_SERVER['DB_NAME'],
   'user' => $_SERVER['DB_USER'],
   'password' => $_SERVER['DB_PASSWORD'],
);

$config = Setup::createAnnotationMetadataConfiguration(array(__DIR__ . '/src/DoctrineEntity'), false);
$config->addEntityNamespace('App', 'Company\Project\DoctrineEntity');
$entity_manager = EntityManager::create($db_params, $config);
$helper_set = ConsoleRunner::createHelperSet($entity_manager);
```

Attempt to use `orm:schema-tool:create` command doesn't create any entities, because instead of resolving `@ORM\Entity` with the above present `use Doctrine\ORM\Mapping as ORM;` line to `Doctrine\ORM\Mapping\Entity` class it resolves it to `Doctrine\ORM\Mapping\ORM\Entity` class. Since it doesn't exist it's being ignored.

This happens likely due fact, that the simple annotation manager has initialized with a hardcoded `Doctrine\ORM\Mapping` namespace and isn't aware of the `ORM` alias under which it's imported.

P.S.
I'm just starting to work with Doctrine ORM, so please excuse my stupid question.

**Comment by beberlei** [comment]
Yes this is how the simple annotation reader works. I believe this is documented, but don't expect any large changes in this, since with PHP 8 attributes, annotations are sort of reaching their end of life

**Reply by aik099** [reply]
How do I get around this if PhpStorm is generating annotations one way, but Doctrine ORM is reading them another way?

Even Doctrine's own documentation has examples with imported `ORM` namespace alias (e.g. https://www.doctrine-project.org/projects/doctrine-annotations/en/1.10/annotations.html), but how do I force Doctrine ORM to read them?

Also at Symfony they somehow made `ORM` namespace alias work (see their entity class: https://symfony.com/doc/current/doctrine.html#creating-an-entity-class), but why Doctrine ORM outside of Symfony doesn't support it.

Thank you.

**Comment by aik099** [comment, marked as answer]
Setting `$useSimpleAnnotationReader` argument of `\Doctrine\ORM\Tools\Setup::createAnnotationMetadataConfiguration` method to `false` did the trick. Now all works as expected.
